tests/test_behavior.py

```diff
@@ -10,17 +12,34 @@
 behaviors: dict = {}
 
 
+class _ClassMethodFn:
+    def __init__(self, attr: str, **kwargs: Any) -> None:
+        self.attr = attr
+
+    def __call__(self, coll: ak.Array, *args: Any, **kwargs: Any) -> ak.Array:
+        return getattr(coll, self.attr)(*args, **kwargs)
+
+
 @ak.mixin_class(behaviors)
 class Point:
-    def distance(self, other):
+    def distance(self, other, __dask_array__=None):
+        if __dask_array__ is not None:
+            return __dask_array__.map_partitions(
+                _ClassMethodFn("distance"),
+                other,
+            )
         return np.sqrt((self.x - other.x) ** 2 + (self.y - other.y) ** 2)
 
     @property
     def x2(self):
```

Comment:
Tests are passing so it appears that everything is working (that is, I remember there being a test for the `x2` property), but according to the docstring you added to `Array.def _call_behavior_property` why don't we need to this to be:

```python
def get_x2(self, __dask_array__=None):
    ...

x2 = property(get_x2)
```

Reply:
Ah! so if `get_x2` is not defined then I assume there is no need for additional calling context and it can be mapped the old way. Needing to define `get_x2` for a property should be opt-in if you need the calling context stuff.

Reply:
Ah OK I'm seeing it now after a re-read, it just falls through to the old `map_partitions` call with _BehaviorPropertyFn, while writing `get_x2` allows the custom-use-of-map_partitions implementation. I'm now understanding the opt-in nature it a bit better! I'll probably ping you to double check some times sometime soon when I'm working on docs

Reply:
I added a commit explaining that a little better.
Additionally all methods get a default calling context of `self`.